modules/RTC/RTCUtils.js

```diff
@@ -1441,6 +1441,26 @@ class RTCUtils extends Listenable {
                         error => reject(error),
                         options);
                 } else if (hasDesktop) {
+                    if (options.screenSharingSourceDeviceId) {
+                        this.getUserMediaWithConstraints(
+                            [ 'video' ],
+                            stream => successCallback({
+                                desktop: {
+                                    sourceId: '',
+                                    screenType: 'camera',
+                                    stream
+                                }
+                            }),
+                            error => reject(error),
+                            {
+                                cameraDeviceId:
+                                    options.screenSharingSourceDeviceId
+                            }
+                        );
+
+                        return;
+                    }
+
                     screenObtainer.obtainStream(
```

Comment:
having an else here (rather than relying on the early return) i think would be more readable/clearer (even if it is redundant).